### PR TITLE
RUBY-2007 Allow mongocrypt handle to take AWS options

### PR DESCRIPTION
**Pull request by egiurleo**
The first of a few PRs adding the AWS KMS option to our FLE implementation.

### CODE REVIEW

File: lib/mongo/crypt/handle.rb [hunk 1]
```diff
@@ -266,6 +266,10 @@ def set_kms_providers_local(kms_providers)
       # Validate and set the aws KMS provider information on the underlying
       # mongocrypt_t object and raise an exception if the operation fails
       def set_kms_providers_aws(kms_providers)
+        unless kms_providers[:aws]
+          raise ArgumentError.new('The :aws KMS provider must not be nil.')
```
Comment (p-mongo) [inline lib/mongo/crypt/handle.rb]: Usually exception messages in Ruby do not have the trailing period. This allows code to chain messages knowing the periods wouldn't be doubled up unnecessarily by following this pattern: `"#{exc}. Some other text"`

File: spec/README.md [hunk 2]
```diff
@@ -287,6 +287,41 @@ Option 1: Download a pre-built binary
...
+```
+
+4. Create a new symmetric Customer Master Key (CMK) by following the "Creating Symmetric CMKs (Console)"
+section of this guide: https://docs.aws.amazon.com/kms/latest/developerguide/create-keys.html
+
+5. Give your IAM user "Key administrator" and "Key user" privileges on your new CMK
+by following the "Using the AWS Management Console Default View" section of this guide:
+https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-modifying.html
+
+TODO: explain where in the test suite to store CMK information (not yet relevant)
```
Comment (egiurleo) [inline spec/README.md]: @p-mongo let me know if this is clear enough. I think linking to AWS documentation is the easiest way to keep this guide up to date, but I'm open to feedback. I haven't explained where in the tests to put the masterkey information, but that will come in a follow-up PR when that info actually gets used in the tests.

File: spec/README.md [hunk 3]
```diff
@@ -287,6 +287,41 @@ Option 1: Download a pre-built binary
 Option 2: Build from source
 - To build libmongocrypt from source, follow the instructions in the README on the libmongocrypt GitHub repo: https://github.com/mongodb/libmongocrypt
 
+Create AWS KMS keys
+Many of the Field-Level Encryption tests require that you have an encryption
+masterkey hosted on AWS's Key Management Service. Set up a masterkey by following
```
Comment (p-mongo) [inline spec/README.md]: Let's use "master key" and `master_key` everywhere we can, i.e. for everything other than calls to the C functions.

File: spec/README.md [hunk 4]
```diff
@@ -287,6 +287,41 @@ Option 1: Download a pre-built binary
...
+"Managing Access Keys (Console)" instructions in this guide:
+https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey
+Once an access key has been created, store the access key id and the access key
+secret in environment variables. If you plan to frequently run Field-Level
+Encryption tests, it may be a good idea to put these lines in your .bash_profile
+or .bashrc file. Otherwise, you can run them in the terminal window where you
+plan to run your tests.
+
+```
+export FLE_AWS_KEY="YOUR-ACCESS-KEY-ID"
```
Comment (p-mongo) [inline spec/README.md]: What do you think about `MONGO_RUBY_DRIVER_AWS_KEY` to match our existing `MONGO_RUBY_DRIVER_LINT` ?

File: spec/README.md (hunk 2 above)
Comment (p-mongo) [reply spec/README.md]: I think this is great.

File: spec/README.md (hunk 4 above)
Comment (egiurleo) [reply spec/README.md]: Sounds good!

**Comment by egiurleo** [comment]
@p-mongo Removed all the evergreen stuff and put some fake AWS credentials in `spec/mongo/crypt/handle_spec.rb` -- it doesn't matter for now that they're not real.

**Pull request merged**